src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -64,10 +138,20 @@ public static TypeSymbolWithAnnotations Create(TypeSymbol typeSymbol, ImmutableA
         {
             if (typeSymbol is null)
             {
-                return null;
+                return default;
             }
 
-            return Create(typeSymbol, typeSymbol.IsNullableType(), customModifiers);
+            return CreateNonLazyType(typeSymbol, NonNullTypesTrueContext.Instance, isAnnotated: typeSymbol.IsNullableType(), customModifiers);
+        }
+
+        internal static TypeSymbolWithAnnotations CreateNonLazyType(TypeSymbol typeSymbol, INonNullTypesContext nonNullTypesContext, bool isAnnotated, ImmutableArray<CustomModifier> customModifiers)
+        {
+            return new TypeSymbolWithAnnotations(typeSymbol, nonNullTypesContext, isAnnotated, Extensions.Create(customModifiers));
+        }
+
+        internal static TypeSymbolWithAnnotations CreateLazyNullableType(CSharpCompilation compilation, TypeSymbolWithAnnotations underlying)
+        {
+            return new TypeSymbolWithAnnotations(defaultType: underlying._defaultType, nonNullTypesContext: underlying.NonNullTypesContext, isAnnotated: true, Extensions.CreateLazy(compilation, underlying));
```

Comment:
>Extensions.CreateLazy(compilation, underlying) [](start = 159, length = 46)

It seems we're still allocating for every TSWA.
To save on allocations, I assumed we'd have `null` extensions for the common case. But that doesn't seem to be the current design. Would that be possible?

Reply:
We're allocating for every `TypeSymbolWithAnnotations` that represent either an annotated type parameter (this case) or a type with associated custom modifiers. For other cases, there shouldn't be allocations.

---
In reply to: [206631539](https://github.com/dotnet/roslyn/pull/28943#discussion_r206631539) [](ancestors = 206631539)